drivers/led/sn32/rgb_matrix_sn32f24xb.c

```diff
@@ -0,0 +1,426 @@
+                cfg->channels[12].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 12;
+                break;
+
+            case C5:
+                cfg->channels[13].pfpamsk = 1;
+            case A13:
+                cfg->channels[13].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 13;
+                break;
+
+            case C6:
+                cfg->channels[14].pfpamsk = 1;
+            case A14:
+                cfg->channels[14].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 14;
+                break;
+
+            case C7:
+                cfg->channels[15].pfpamsk = 1;
+            case A15:
+                cfg->channels[15].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 15;
+                break;
+
+            case C8:
+                cfg->channels[16].pfpamsk = 1;
+            case B0:
+                cfg->channels[16].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 16;
+                break;
+
+            case C9:
+                cfg->channels[17].pfpamsk = 1;
+            case B1:
+                cfg->channels[17].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 17;
+                break;
+
+            case C10:
+                cfg->channels[18].pfpamsk = 1;
+            case B2:
+                cfg->channels[18].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 18;
+                break;
+
+            case C11:
+                cfg->channels[19].pfpamsk = 1;
+            case B3:
+                cfg->channels[19].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 19;
+                break;
+
+            case C12:
+                cfg->channels[20].pfpamsk = 1;
+            case B4:
+                cfg->channels[20].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 20;
+                break;
+
+            case C13:
+                cfg->channels[21].pfpamsk = 1;
+            case B5:
+                cfg->channels[21].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 21;
+                break;
+
+            case C14:
+                cfg->channels[22].pfpamsk = 1;
+            case B6:
+                cfg->channels[22].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 22;
+                break;
+
+            case C15:
+                cfg->channels[23].pfpamsk = 1;
+            case B7:
+                cfg->channels[23].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i] = 23;
+                break;
+        }
+    }
+}
+void rgb_callback(PWMDriver *pwmp);
+
+void shared_matrix_rgb_enable(void) {
+    pwmcfg.callback = rgb_callback;
+    pwmEnablePeriodicNotification(&PWMD1);
+}
+
+void shared_matrix_rgb_disable_pwm(void) {
+    // Disable PWM outputs on column pins
+    for(uint8_t y = 0; y < LED_MATRIX_COLS; y++) {
+        pwmDisableChannel(&PWMD1,chan_col_order[y]);
```

Comment:
```suggestion
        pwmDisableChannelI(&PWMD1,chan_col_order[y]);
```
(this is already called from inside the lock zone).